pkg/apis/v1alpha1/subnet_types.go

```diff
@@ -0,0 +1,75 @@
+/* Copyright © 2021 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type SubnetStatusCondition string
+
+const (
+	SubnetReady    SubnetStatusCondition = "Ready"
+	SubnetNotReady SubnetStatusCondition = "NotReady"
+)
+
+// SubnetSpec defines the desired state of Subnet
+type SubnetSpec struct {
+	// Size of subnet based upon estimated workload count.
+	// Defaults to 64.
+	// +kubebuilder:default:=64
+	IPV4SubnetSize string `json:"ipv4SubnetSize"`
+	// Access mode of subnet, accessible only from within VPC or from outside VPC.
+	// Defaults to private.
+	// +kubebuilder:default:=private
+	AccessMode string `json:"accessMode"`
```

Comment:
Could we also add "+kubebuilder:validation:Enum" to limit it can only be public or private?

Reply:
Thanks, also add `isolated` in enum, not sure whether `isolate` is needed.